### DISCUSSION TITLE
style 2 protein models with different colors

### DISCUSSION THREAD

**Discussion by lee-t** [swsoyee/r3dmol#55] [Q&A] [answered]
Is there a way to color or style 2 different models? Say for the minimal example below, I want to color model `a` with one color and then add another model `b` and color that one differently. In the code below, both protein models will be orange, because `m_set_style` is global and accepts only atom selections. There does not seem to be an obvious way to style the 2nd model or select only the atoms of the 2nd model. `m_sel` seems to have `model=` as an option but there doesnt seem to be a place to name your model with `m_add_model` and I've tried various array and numeric accessors. 

```{r}
library(bio3d)
library(r3dmol)

# read two G-protein structures
a <- read.pdb("4q21")
b <- read.pdb("4lhy")

r3dmol() %>%
     m_add_model(data = m_bio3d(a)) %>%
  m_set_style(                  
    style = m_style_cartoon(
      color = "aquamarine"
    )) %>%
    m_add_model(data =  m_bio3d(b)) %>%
      m_set_style(                  
    style = m_style_cartoon(
      color = "orange"
    )) %>%
    m_zoom_to() 
```

**Comment by swsoyee** [comment, marked as answer]
Hi, you could try `m_sel()` to color them by the model index.

```r
library(bio3d)
library(r3dmol)

# read two G-protein structures
a <- read.pdb("4q21")
b <- read.pdb("4lhy")

r3dmol() %>%
  m_add_model(data = m_bio3d(a)) %>%
  m_set_style(
    sel = m_sel(model = 0),
    style = m_style_cartoon(
      color = "aquamarine"
    )) %>%
  m_add_model(data =  m_bio3d(b)) %>%
  m_set_style(
    sel = m_sel(model = 1),
    style = m_style_cartoon(
      color = "orange"
    )) %>%
  m_zoom_to() 
```


<img width="527" alt="2024-10-29 16 11 00" src="https://github.com/user-attachments/assets/43c6833f-9725-402e-960f-bae4e865ff51">
